iree/compiler/Dialect/Flow/Transforms/DispatchLinalgOnTensors.cpp

```diff
@@ -933,45 +949,45 @@ struct MakeDispatchWorkgroupsOp : public RewritePattern {
 /// heuristic is used below, but the mechanism should be general enough to
 /// capture any heuristic.
 
-/// Checks if the `producer` can be fused into the dispatch region of the
-/// `consumer`.
-static bool isProducerFusable(linalg::LinalgOp producer,
-                              linalg::LinalgOp consumer,
-                              OpOperand &consumerOperand) {
-  if (consumer.isInputTensor(&consumerOperand)) {
-    if (!clEnableOperandFusion) return false;
-
-    // Make sure that we have an identity indexing map for the operand for now.
-    //
-    // Theoretically with tensor abstraction, we can pull in whatever operand
-    // subrange for fusion. But if the consumer's access patterns for the input
-    // and output are not the same, we must materialize buffers for holding the
-    // temporary result, because we can have multiple levels of distribution.
-    // Identity map makes sure that we can actually avoid a new intermediate
-    // buffer and directly use the one for the consumer.
-    auto map = consumer.getIndexingMap(consumerOperand.getOperandNumber());
-    if (!map.isIdentity()) return false;
-
-    // If the producer's result is used by the consumer as an input, fuse if
-    // this producer has no other users.
-    return consumerOperand.get().hasOneUse();
-  } else {
-    // If the producer's result is used by the consumer as an output
-    // initializer, fuse if the producer is an elementwise parallel operation.
-    auto isElementWiseParallelOp = [](linalg::LinalgOp op) {
-      return llvm::all_of(op.iterator_types(), [](Attribute attr) {
-        return linalg::isParallelIteratorType(attr);
-      });
-    };
-    return isElementWiseParallelOp(producer);
+/// Method to set elementwise operations as root operations.
+// TODO(#5045): After the regression issue on CPU side is addressed, this can be
+// folded into the main logic of fusion.
+template <typename GenericOpTy>
```

Comment:
Is it really necessary to template this? FWICT, you can just check `isa<GenericOp, IndexedGenericOp>()`, right?

Reply:
unfortunately yes. It seems like the tile+distribute was done only for generic operations and not indexed generic operations. So I am keeping the same functionality here for the GPU side of things.
So when tile+distribute of elementwise operations is enabled its done only for generic ops. But after that to use the fallback dispatch region creation you need to make both indexed generic and generic ops root ops, so that fill ops can be fused with them.

Reply:
Ah, right. I recall now. Thanks.